v2/cli/init_command.go

```diff
@@ -0,0 +1,255 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/brigadecore/brigade/v2/internal/file"
+	"github.com/urfave/cli/v2"
+)
+
+type initFields struct {
+	ProjectID   string
+	Language    string
+	GitCloneURL string
+	Script      string
+}
+
+var initCommand = &cli.Command{
+	Name:  "init",
+	Usage: "Initialize a new Brigade project",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:    flagID,
+			Aliases: []string{"i"},
+			Usage: "Name of the Brigade project to initialize " +
+				"(required)",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:    flagLanguage,
+			Aliases: []string{"l"},
+			Usage: "Specify the scripting language to use for handling events" +
+				" -- JavaScript (js) or TypeScript (ts)",
+			Value: "ts",
+		},
+		&cli.StringFlag{
+			Name:    flagGit,
+			Aliases: []string{"g"},
+			Usage: "Optionally specify a git repository where the" +
+				" event handling script can be found",
+		},
+	},
+	Action: initialize,
+}
+
+// Handle init CLI flags
+// nolint: gocyclo
+func initialize(c *cli.Context) error {
+
+	// Initialize fields to fill in the template yaml files
+	fields := initFields{
+		ProjectID:   c.String(flagID),
+		Language:    strings.ToLower(c.String(flagLanguage)),
+		GitCloneURL: c.String(flagGit),
+	}
+
+	err := idMatch(fields.ProjectID)
+	if err != nil {
+		return err
+	}
+
+	// Check if input language is valid
+	fields.Language, err = canonicalizeLanguage(fields.Language)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade directory exists
+	err = checkCreateDirectory(".brigade")
+	if err != nil {
+		return err
+	}
+
+	// Generate js/ts buffer
+	var scriptBytes []byte
+	scriptBytes, err = execTemplate(scriptTemplate, fields)
+	if err != nil {
+		return err
+	}
+	fields.Script = string(scriptBytes)
+
+	brigadeScriptFileName := fmt.Sprintf("brigade.%s", fields.Language)
+
+	// If git repo is provided, create a separate js/ts file for script
+	var overwrite bool
+	scriptPath := path.Join(".brigade", brigadeScriptFileName)
+	if fields.GitCloneURL != "" {
+		if overwrite, err = checkFileOverwrite(scriptPath); err != nil {
+			return err
+		} else if overwrite {
+			if err = ioutil.WriteFile(
+				scriptPath,
+				scriptBytes,
+				0644,
+			); err != nil {
+				return err
+			}
+		}
+	}
+
+	// Script is added to project.yaml otherwise
```

Comment:
This comment is misleading because it talks about something that happens conditionally, but there's no obvious conditional logic here. I _do_ happen to know that the conditional is inside the template, but don't assume someone new who comes along and reads this code will posses that degree of knowledge or context.

It think, in this case, it's clearer to just remove this comment.

Reply:
Or rewrite the comment to just say something like "Generate the project.yaml file," without mention of conditions that are actually buried within the template.